### PR TITLE
style(replays): increase z-index of network tab close buton

### LINKED ISSUE #59566: Replay: bottom Network pane actions access issues
### Environment

SaaS (https://sentry.io/)

### Steps to Reproduce

1. Pick a Replay
2. Open the Network tab
3. Select a request
4. Try to access the actions (Request, Response, X - especially the X button)
5. If you try to access it "horizontally" the "drag to resize" state kicks in and it removes the actions. You have to access it from "below".

### Expected Result

The "drag to resize" state shouldn't remove the actions.

### Actual Result

The actions disappear while the "drag to resize" state is present.

https://github.com/getsentry/sentry/assets/5396211/80992371-0b40-429c-a5f7-9592958ee2df

### Product Area

Replays

### Link

> Not applicable

### DSN

https://af6bbb2a2670282fec7d94ad6d64872f@o4506044970565632.ingest.sentry.io/4506185944989697

### Version

> SaaS

### PR DESCRIPTION
**Pull request by michellewzhang**
fixes https://github.com/getsentry/sentry/issues/59566. the z-index of the close button is now higher, so it'll always be visible, even if the split divider is hovered

new behavior:

https://github.com/getsentry/sentry/assets/56095982/fd127d45-0978-4ae3-9a07-2c21f8f89a44

(old behavior for reference):

https://github.com/getsentry/sentry/assets/56095982/4c94d017-ef19-4604-b51d-353ed1c96665

### CODE REVIEW

**Review by JoshFerge — APPROVED** [review]

**Review by billyvg — COMMENTED** [review]
Instead of changing z-index, can we remove the float of the close button? Right now the hitbox of the resize bar is:

```
| Details | Request | Response | <-[:::]--------- resize -----------[x]->|
```


If we removed the float and changed some styling around, we could make the close button behave like the tabs (and even adjust the hitbox if needed:

```
| Details | Request | Response | <-[:::]--------- resize ------->  [x]|
```

e.g.

```
| Details | Request | Response | <-[:::]----->            [x]|
```

**Review by c298lee — APPROVED** [review]

**Comment by billyvg** [comment]
After playing with it a bit, I like having the close button on top, gives users the ability to close it instead of feeling kind of lost of why the bar disappeared.

File: static/app/views/replays/detail/network/details/index.tsx [hunk 1]
```diff
@@ -113,8 +113,8 @@ const CloseButtonWrapper = styled('div')`
   position: absolute;
   right: 0;
   height: 100%;
-  padding: ${space(1)};
-  z-index: ${p => p.theme.zIndex.initial};
+  padding: ${space(2)};
```
Comment (billyvg) [inline static/app/views/replays/detail/network/details/index.tsx:116]: Intentional?

**Review by billyvg — APPROVED** [review]

File: static/app/views/replays/detail/network/details/index.tsx (hunk 1 above)
Comment (michellewzhang) [reply static/app/views/replays/detail/network/details/index.tsx]: you were tooooo quick. just removed

**Pull request merged**